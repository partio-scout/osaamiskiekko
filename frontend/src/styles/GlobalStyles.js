import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
@import url('https://fonts.googleapis.com/css?family=Fira+Sans:400,400i,700,700i|Nunito+Sans:900&display=swap');
@import url("//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css");

body {
  font-family: "Fira Sans", sans-serif;
  margin: 0px;
  padding: 0px;
  background-color: ${props => props.theme.colors.backgroundPrimary}
  color: #202020;
}

h1, h1 span {
  color: #202020;
  font-family: "Nunito Sans", sans-serif;
  font-weight: 900;
  font-size: 48px
  line-height: 52px;
}

h2, h2 span {
  color: #202020;
  font-family: "Fira Sans", sans-serif;
  font-weight: bold;
  font-size: 33px
  line-height: 36px;
}

h3, h3 span {
  color: #202020;
  font-family: "Fira Sans", sans-serif;
  font-weight: bold;
  font-size: 26px
  line-height: 28px;
}

p.lead {
  color: #202020;
  font-family: "Fira Sans", sans-serif;
  font-size: 20px
  line-height: 30px;
}

label, span.label {
  color: #202020;	
  font-family: "Fira Sans", sans-serif;
  font-size: 16px;
  letter-spacing: 2.29px;
  line-height: 15px;
}

label.secondary, span.label.secondary {
  color: #6E6E6E;
  font-family: "Fira Sans", sans-serif;
  font-size: 16px;
  letter-spacing: 2.29px;
  line-height: 15px;
}

.p {
	color: #202020;
  font-family: "Fira Sans", sans-serif;
	font-size: 18px;
	line-height: 27px;
}

@media only screen and (max-width: 730px) {
  h1, h1 span {
    color: #202020;
    font-family: "Nunito Sans", sans-serif;
    font-weight: 900;
    font-size: 37px
    line-height: 36px;
  }
  
  h2, h2 span {
    color: #202020;
    font-family: "Fira Sans", sans-serif;
    font-weight: bold;
    font-size: 26px
    line-height: 32px;
  }
  
  h3, h3 span {
    color: #202020;
    font-family: "Fira Sans", sans-serif;
    font-weight: bold;
    font-size: 20px
    line-height: 28px;
  }
  
  p.lead {
    color: #202020;
    font-family: "Fira Sans", sans-serif;
    font-size: 18px
    line-height: 26px;
  }
  
  label, span.label {
    color: #202020;	
    font-family: "Fira Sans", sans-serif;
    font-size: 14px;
    letter-spacing: 2px;
    line-height: 15px;
  }
  
  label.secondary, span.label.secondary {
    color: #6E6E6E;
    font-family: "Fira Sans", sans-serif;
    font-size: 12px;
    letter-spacing: 2px;
    line-height: 15px;
  }
  
  .p {
    color: #202020;
    font-family: "Fira Sans", sans-serif;
    font-size: 16px;
    line-height: 24px;
  }
}
`;

export default GlobalStyle;
