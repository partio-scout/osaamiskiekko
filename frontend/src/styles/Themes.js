const defaultTheme = {
  colors: {
    osaamisKiekkoBlue: '#00283B',
    heading: '#00283B',
    white: '#FFFFFF',
    osaamisKiekkoSand: '#E8EAE3'
  },
  fonts: {
    headingFontFamily: 'Futura'
  }
}

const darkTheme = {
  colors: {
    osaamisKiekkoBlue: '#30312B',
    osaamisKiekkoSand: '#bac6bd'
  },
  fonts: {
    headingFontFamily: 'Futura'
  }
}

export { defaultTheme, darkTheme };