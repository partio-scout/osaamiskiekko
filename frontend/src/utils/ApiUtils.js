export const addTypeToSchoolOrOrganization = (schoolOrOrganization, type) => schoolOrOrganization.map(item => {
    if (type === 'school') {
      const o = Object.assign({}, item);
      o.type_fi = 'Oppilaitos';
      o.type_en = 'School';
      o.type_se = 'Skola'
      return o;
    }
    return null;
  })