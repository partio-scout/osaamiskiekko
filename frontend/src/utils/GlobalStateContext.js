import React from 'react';
// import { defaultTheme } from '../styles/Themes';

export const useGlobalStateContext = () => React.useContext(GlobalStateContext);

const defaultValues = {
  language: 'fi',
  changeLanguage: (language) => {  console.log(language) },
  currentTheme: 'defaultTheme',
  changeCurrentTheme: (theme) => { console.log(theme) },
  isLoading: true,
  error: undefined,
  nqfLevels: undefined,
  fieldOfStudies: undefined,
  schools: undefined, 
  organizations: undefined,
};

const GlobalStateContext = React.createContext(defaultValues);

export default GlobalStateContext;
