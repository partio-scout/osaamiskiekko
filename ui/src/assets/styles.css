.container {
    background-image: linear-gradient(
        to bottom,
        rgb(var(--v-theme-bg-gradient-start)),
        rgb(var(--v-theme-bg-gradient-end))
    );
}
/* TEXT */
.dm-sans {
    font-family: 'DM Sans', sans-serif !important;
}
.dm-sans-white {
    color: rgb(var(--v-theme-h2h3));
    font-family: 'DM Sans', sans-serif !important;
}
.poppins {
    font-family: 'Poppins', sans-serif !important;
}
.poppins-white {
    color: rgb(var(--v-theme-h2h3));
    font-family: 'Poppins', sans-serif !important;
}
.poppins-purple {
    color: rgb(var(--v-theme-purple));
    font-family: 'Poppins', sans-serif !important;
}
.poppins-turquoise {
    color: rgb(var(--v-theme-turquoise));
    font-family: 'Poppins', sans-serif !important;
}
.heading {
    color: rgb(var(--v-theme-h1));
    font-family: 'Poppins', sans-serif !important;
}
/* HYPHENATION & WORD BREAK RULES*/
.hyphenate-break-5 {
    display: -webkit-box;
    -webkit-line-clamp: 5; /* number of lines to show */
    -webkit-box-orient: vertical;
    word-break: break-word;
    hyphens: auto;
    overflow: hidden;
}
.hyphenate-break-3 {
    display: -webkit-box;
    -webkit-line-clamp: 3; /* number of lines to show */
    -webkit-box-orient: vertical;
    word-break: break-word;
    hyphens: auto;
    overflow: hidden;
}
.hyphenate-break-2 {
    display: -webkit-box;
    -webkit-line-clamp: 2; /* number of lines to show */
    -webkit-box-orient: vertical;
    word-break: break-word;
    hyphens: auto;
    overflow: hidden;
}
.hyphenate-break-1 {
    display: -webkit-box;
    -webkit-line-clamp: 1; /* number of lines to show */
    -webkit-box-orient: vertical;
    word-break: break-word;
    hyphens: auto;
    overflow: hidden;
}
/* BUTTONS */
.button {
    width: 300px;
    height: 60px !important;
    border-radius: 30px !important;
    background: transparent;
    border: 4px solid rgb(var(--v-theme-blue));
    color: rgb(var(--v-theme-blue));
    font-family: 'Poppins', sans-serif !important;
    font-size: medium;
    transition: color 0.6s;
    transition: background 0.6s;
}
.button:hover {
    background: rgb(var(--v-theme-blue));
    color: rgb(var(--v-theme-bg-gradient-end));
}
.search-button {
    width: 130px;
    height: 50px !important;
    border-radius: 30px !important;
    background: rgb(var(--v-theme-blue));
    color: rgb(var(--v-theme-h2h3));
    font-family: 'Poppins', sans-serif !important;
    font-size: larger;
}
.button--turquoise {
    width: 300px;
    height: 60px !important;
    border-radius: 30px !important;
    background: transparent;
    border: 4px solid rgb(var(--v-theme-turquoise));
    color: rgb(var(--v-theme-turquoise));
    font-family: 'Poppins', sans-serif !important;
    font-size: medium;
}
.button--yellow {
    width: 300px;
    height: 60px !important;
    border-radius: 30px !important;
    background: transparent;
    border: 4px solid rgb(var(--v-theme-yellow));
    color: rgb(var(--v-theme-yellow));
    font-family: 'Poppins', sans-serif !important;
    font-size: medium;
}
.button--purple {
    width: 300px;
    height: 60px !important;
    border-radius: 30px !important;
    background: transparent;
    border: 4px solid rgb(var(--v-theme-purple));
    color: rgb(var(--v-theme-purple));
    font-family: 'Poppins', sans-serif !important;
    font-size: medium;
}
/* CARDS */
.degree-card {
    border-radius: 30px;
    background-color: rgb(var(--v-theme-turquoise));
    transition: transform 0.6s;
    /* transition: background-color 0.6s; */
}
.degree-card--no-hover {
    border-radius: 30px;
    background-color: rgb(var(--v-theme-turquoise));
}
.degree-card:hover {
    /* border-radius: 30px; */
    /* background-color: #bafff0; */
    transform: scale(1.06);
}
.yellow {
    background-color: rgb(var(--v-theme-yellow));
}
/* .yellow:hover {
    background-color: #FBFFB9;
} */
.yellow--no-hover {
    background-color: rgb(var(--v-theme-yellow));
}
.skill-card {
    border-radius: 30px;
    background-color: rgb(var(--v-theme-purple));
    transition: transform 0.6s;
    /* transition: background-color 0.6s; */
}
.skill-card--no-hover {
    border-radius: 30px;
    background-color: rgb(var(--v-theme-purple));
}
.skill-card:hover {
    /* border-radius: 30px; */
    /* background-color: #F2DAF6; */
    transform: scale(1.06);

}
.result-card {
    border-radius: 30px;
    background-color: rgb(var(--v-theme-h1));
}
.card-footer {
    background-color: rgb(var(--v-theme-h1));
    border-top-left-radius: 30px;
}
.card-footer-transparent {
    opacity: 40%;
}
.card-footer-wide {
    background-color: rgb(var(--v-theme-h2h3));
}