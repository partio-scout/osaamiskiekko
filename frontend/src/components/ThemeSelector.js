import React from 'react';
import styled from 'styled-components';
import { useGlobalStateContext } from '../utils/GlobalStateContext';
import { themes, findTheme, defaultTheme } from '../styles/Themes';

const S = {};
S.themeSelector = styled.div`
position: relative;

select {
  height: 49px;	
  width: 160px;	
  border-radius: 24.5px;	
  background-color: ${props => props.theme.colors.highlight};
  color: ${props => props.theme.colors.highlightText};
  padding: 5px 5px 5px 20px;
  border: 2px solid ${props => props.theme.colors.highlight};;
  font-size: 14px;
  -webkit-appearance: none;
  appearance: none;
  outline: none;
  text-transform: uppercase;
  font-weight: bold;
}

select:focus {
  border: 2px solid ${props => props.theme.colors.highlightText};
}

::before {
  content: "\f0d7";
  font-family: FontAwesome;
  position: absolute;
  font-size: 5px;
  top: 0;
  right: 10px;
  width: 20%;
  height: 100%;
  text-align: center;
  font-size: 15px;
  line-height: 50px;
  pointer-events: none;
  color: ${props => props.theme.colors.highlightText};
  :hover {
    color: rgba(0, 0, 0, 0.6);
    background-color: rgba(255, 255, 255, 0.2);
  }
}

select option {
  padding: 30px;
}
`;

export default function Navigation() {
  const { changeTheme } = useGlobalStateContext();
  const setTheme = (e) => {
    const name = e.target.value;

    changeTheme(findTheme(name) || defaultTheme);
  }

  return (
    <S.themeSelector>
      <select onChange={setTheme}>
        {themes.map(theme => { return (
          <option key={theme.name} value={theme.name}>
            {theme.name}
          </option>
        )})}
      </select>
    </S.themeSelector>
  )
}
