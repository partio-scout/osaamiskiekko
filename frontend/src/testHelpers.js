import React from 'react';
import { IntlProvider, addLocaleData } from 'react-intl';
import { ThemeProvider } from 'styled-components';

import GlobalStateContext from './utils/GlobalStateContext';
import fi from 'react-intl/locale-data/fi';
import translations from './translations/translations';
import { defaultTheme } from './styles/Themes';

addLocaleData([...fi, translations]);

export default function renderWithProviders(component, globalState) {

  return (
    <GlobalStateContext.Provider value={globalState}>
      <ThemeProvider theme={defaultTheme}>
        <IntlProvider locale={'fi'} messages={translations['fi']}>
          {component}
        </IntlProvider>
      </ThemeProvider>
    </GlobalStateContext.Provider>
  );
};
