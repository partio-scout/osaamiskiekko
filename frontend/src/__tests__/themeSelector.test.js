import React from 'react';
import {mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ThemeSelector from '../components/ThemeSelector';
import * as GlobalStateContext from '../utils/GlobalStateContext';
import { themes, defaultTheme, findTheme } from '../styles/Themes';
import { wrapWithProviders } from '../testHelpers';

configure({adapter: new Adapter()});

describe('themes', () => {

  test('can find green theme', () => {
    const found = findTheme('green');

    expect(found.name).toBe('green');
  });

  test('returns default theme if non-existent theme is sought', () => {
    const found = findTheme('iamsononexistentreallyanunicorn');

    expect(found.name).toBe('blue');
  });
});

describe('theme selector', () => {
  test('lists all themes', () => {
    const wrapper = mount(wrapWithProviders(
      <ThemeSelector />
    ));
    expect(wrapper.find('option').length).toBe(themes.length);
  });

  test('sets theme to global state', () => {
    const mockChangeTheme = jest.fn(theme => { globalValues.theme = theme });
    const globalValues = {
      theme: defaultTheme,
      changeTheme: mockChangeTheme
    };
    jest.spyOn(GlobalStateContext, 'useGlobalStateContext')
      .mockImplementation(() => globalValues);

    const wrapper = mount(wrapWithProviders(
      <ThemeSelector />
    ));

    wrapper.find('select').simulate('change', {target: {value : 'green'}});

    expect(mockChangeTheme.mock.calls.length).toBe(1);
    expect(globalValues.theme.name).toBe('green');
  });
});
