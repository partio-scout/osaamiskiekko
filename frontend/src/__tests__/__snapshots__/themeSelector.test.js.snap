// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme selector snapshot matches 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "backgroundPrimary": "#ffffff",
        "backgroundSecondary": "#d6e0e5",
        "highlight": "#4300ee",
        "highlightText": "#ffffff",
        "link": "#4300ee",
        "text": "#262626",
      },
      "fonts": Object {
        "headingFontFamily": "Futura",
      },
      "name": "blue",
    }
  }
>
  <IntlProvider
    locale="fi"
    messages={
      Object {
        "header.h1": "Osaamiskiekon avulla hyödynnät osaamisesi monipuolisesti",
        "header.important": "Osaamiskiekko on vielä kehitteillä ja avataan kokonaisena vasta kesällä 2019!",
        "header.p": "Osaamiskiekon avulla löydät harrastus- ja vapaaehtoistoiminnassa hankitun osaamisen rinnakkain tutkintoon johtavien opintojen kanssa.",
        "nav.changetheme": "Valitse teema",
        "nav.language.en": "Change language to English",
        "nav.language.fi": "Vaihda kieli suomeksi",
        "nav.language.sv": "Byt språk till svenska",
        "nav.osaamiskiekko": "Osaamiskiekko",
        "nav.otayhteytta": "Ota yhteyttä",
        "nav.tietoa": "Tietoa",
        "search.placeholder": "Hae...",
        "searchbox.label": "Valitse järjestö tai oppilaitos",
        "searchbox.labelSecondary": "Valitse koulutus",
        "themes.blue": "Sinisikermä",
        "themes.green": "Sammal",
        "themes.grey": "Harmahdus",
      }
    }
  >
    <InjectIntl(ThemeSelector)>
      <ThemeSelector
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "fi",
            "messages": Object {
              "header.h1": "Osaamiskiekon avulla hyödynnät osaamisesi monipuolisesti",
              "header.important": "Osaamiskiekko on vielä kehitteillä ja avataan kokonaisena vasta kesällä 2019!",
              "header.p": "Osaamiskiekon avulla löydät harrastus- ja vapaaehtoistoiminnassa hankitun osaamisen rinnakkain tutkintoon johtavien opintojen kanssa.",
              "nav.changetheme": "Valitse teema",
              "nav.language.en": "Change language to English",
              "nav.language.fi": "Vaihda kieli suomeksi",
              "nav.language.sv": "Byt språk till svenska",
              "nav.osaamiskiekko": "Osaamiskiekko",
              "nav.otayhteytta": "Ota yhteyttä",
              "nav.tietoa": "Tietoa",
              "search.placeholder": "Hae...",
              "searchbox.label": "Valitse järjestö tai oppilaitos",
              "searchbox.labelSecondary": "Valitse koulutus",
              "themes.blue": "Sinisikermä",
              "themes.green": "Sammal",
              "themes.grey": "Harmahdus",
            },
            "now": [Function],
            "onError": [Function],
            "textComponent": "span",
            "timeZone": null,
          }
        }
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "fHIxWH",
                  "rules": Array [
                    "
position: relative;

select {
  height: 49px;	
  border-radius: 24.5px;	
  background-color: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  padding: 5px 40px 5px 20px;
  border: 2px solid ",
                    [Function],
                    ";;
  font-size: 14px;
  -webkit-appearance: none;
  appearance: none;
  outline: none;
  text-transform: uppercase;
  font-weight: bold;
}

select:focus {
  border: 2px solid ",
                    [Function],
                    ";
}

::before {
  content: \\"0d7\\";
  font-family: FontAwesome;
  position: absolute;
  font-size: 5px;
  top: 0;
  right: 10px;
  width: 10px;
  height: 100%;
  text-align: center;
  font-size: 15px;
  line-height: 50px;
  pointer-events: none;
  color: ",
                    [Function],
                    ";
  :hover {
    color: rgba(0, 0, 0, 0.6);
    background-color: rgba(255, 255, 255, 0.2);
  }
}

select option {
  padding: 30px;
}
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bdVaJa fHIxWH"
            >
              <select
                aria-label="Valitse teema"
                onChange={[Function]}
              >
                <option
                  key="blue"
                  value="blue"
                >
                  Sinisikermä
                </option>
                <option
                  key="green"
                  value="green"
                >
                  Sammal
                </option>
                <option
                  key="grey"
                  value="grey"
                >
                  Harmahdus
                </option>
              </select>
            </div>
          </StyledComponent>
        </styled.div>
      </ThemeSelector>
    </InjectIntl(ThemeSelector)>
  </IntlProvider>
</ThemeProvider>
`;
